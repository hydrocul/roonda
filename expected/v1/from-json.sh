#!/bin/sh

#################################################
# main.sh:
#################################################
cat <<\END_OF_ROONDA_SOURCE_FILE > $ROONDA_TMP_PATH/main.sh

cd $ROONDA_TMP_PATH

cat source.json | $ROONDA_SELF_PATH --from-json --dry-run

cat source.json | $ROONDA_SELF_PATH --from-json

END_OF_ROONDA_SOURCE_FILE
#################################################
# source.json:
#################################################
cat <<\END_OF_ROONDA_SOURCE_FILE > $ROONDA_TMP_PATH/source.json
["sh", "v1",
 ["echo", "Hello"],
 ["echo", "Hello", "World"],
 ["echo", "Hello   World!"]]
END_OF_ROONDA_SOURCE_FILE
#################################################

sh $ROONDA_TMP_PATH/main.sh
