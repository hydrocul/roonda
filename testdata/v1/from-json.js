["sh", "v1",
 ["echo", "Hello"],
 ["echo", "Hello", "World"],
 ["echo", "Hello   World!"]]
