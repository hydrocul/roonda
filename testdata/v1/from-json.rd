sh v1

("sh" main.sh)
# TODO "sh" のように文字列にしないとエラーになってしまう

<< main.sh

cd $ROONDA_TMP_PATH

cat source.json | $ROONDA_SELF_PATH --from-json --dry-run

cat source.json | $ROONDA_SELF_PATH --from-json

main.sh >>

<< source.json
["sh", "v1",
 ["echo", "Hello"],
 ["echo", "Hello", "World"],
 ["echo", "Hello   World!"]]
source.json >>
