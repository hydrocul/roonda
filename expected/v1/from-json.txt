#!/bin/sh

echo Hello
echo Hello World
echo 'Hello   World!'
Hello
Hello World
Hello   World!
